src/blob/index/bptree/node.rs

```diff
@@ -0,0 +1,103 @@
+use super::prelude::*;
+
+#[derive(Debug, PartialEq)]
+pub(super) struct Node {
+    keys: Vec<Vec<u8>>,
+    offsets: Vec<u64>,
+}
+
+impl Node {
+    pub(super) fn new(keys: Vec<Vec<u8>>, offsets: Vec<u64>) -> Self {
+        assert!(keys.len() != 0);
+        assert_eq!(keys.len() + 1, offsets.len());
+        Self { keys, offsets }
+    }
+
+    pub(super) fn key_offset(&self, key: &[u8]) -> u64 {
+        match self.keys.binary_search_by(|elem| elem.as_slice().cmp(key)) {
+            Ok(pos) => self.offsets[pos + 1],
+            Err(pos) => self.offsets[pos],
+        }
+    }
+
+    pub(super) fn serialize(&self) -> Result<Vec<u8>> {
+        let mut buf = Vec::new();
+        let meta = NodeMeta::new(self.keys.len() as u64);
+        buf.extend_from_slice(&serialize(&meta)?);
+        buf.extend(self.keys.iter().flat_map(|kb| kb));
+        self.offsets.iter().try_fold(buf, |mut acc, offset| {
+            acc.extend_from_slice(&serialize(offset)?);
+            Ok(acc)
+        })
+    }
+
+    pub(super) fn deserialize(buf: &[u8], key_size: u64) -> Result<Self> {
+        let meta_size = NodeMeta::serialized_size_default()?;
+        let (meta_buf, data_buf) = buf.split_at(meta_size as usize);
+        let meta: NodeMeta = deserialize(&meta_buf)?;
+        let keys_buf_size = meta.size * key_size;
+        let (keys_buf, rest_buf) = data_buf.split_at(keys_buf_size as usize);
+        let keys = keys_buf
+            .chunks(key_size as usize)
+            .map(|key| key.to_vec())
+            .collect();
+        let offsets_buf_size = (meta.size + 1) as usize * std::mem::size_of::<u64>();
+        let (offsets_buf, _rest_buf) = rest_buf.split_at(offsets_buf_size);
+        let offsets = offsets_buf
+            .chunks(std::mem::size_of::<u64>())
+            .map(|bytes| deserialize::<u64>(bytes).map_err(Into::into))
+            .collect::<Result<Vec<u64>>>()?;
```

Comment:
Maybe it's more convenient to use here `try_collect()`

Reply:
There is an error: this method doesn't exists for `Map<Chunks<..>`, because this struct doesn't implement `TryStream` trait (because it's an iterator; we can use `itertools` crate to use this method for iterator, but that's redundant imo)